rand.go

```diff
@@ -0,0 +1,47 @@
+package retry
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var r = &lockedSource{src: rand.New(rand.NewSource(time.Now().UnixNano()))}
```

Comment:
Since we're creating our own instance, I think we should not have a global. We can create a unique `r` inside `WithJitter`, outside the first closure.